docker/service/src/main/java/uk/nhs/adaptors/scr/fhirmappings/PractitionerRoleMapper.java

```diff
@@ -0,0 +1,76 @@
+package uk.nhs.adaptors.scr.fhirmappings;
+
+import static uk.nhs.adaptors.scr.utils.ThrowFhirMappingError.throwFhirMappingError;
+import static uk.nhs.adaptors.scr.utils.ThrowFhirMappingError.throwFhirMappingErrorMissingFromPayload;
+
+import org.hl7.fhir.r4.model.PractitionerRole;
+
+import uk.nhs.adaptors.scr.exceptions.FhirMappingException;
+import uk.nhs.adaptors.scr.models.GpSummary;
+
+public class PractitionerRoleMapper {
+
+    public GpSummary mapPractitionerRole(GpSummary gpSummary, PractitionerRole practitionerRole) throws FhirMappingException {
+        if (practitionerRole == null) {
+            throwFhirMappingErrorMissingFromPayload("PractitionerRole");
+        }
+        setPractitionerRoleId(gpSummary, practitionerRole);
+        setPractitionerRoleCodeAndCodeDisplayName(gpSummary, practitionerRole);
+        setPractitionerRoleIsAgentPersonSDS(gpSummary, practitionerRole);
+
+        return gpSummary;
+    }
+
+    private static void setPractitionerRoleId(GpSummary gpSummary, PractitionerRole practitionerRole) throws FhirMappingException {
+        String value = "";
+
+        if (practitionerRole.hasIdentifier()) {
+            if (practitionerRole.getIdentifierFirstRep().hasValue()) {
+                value = practitionerRole.getIdentifierFirstRep().getValue();
+            }
+        }
+
+        gpSummary.setPractitionerRoleId(value);
+    }
+
+    private static void setPractitionerRoleCodeAndCodeDisplayName(GpSummary gpSummary, PractitionerRole practitionerRole)
+        throws FhirMappingException {
+        String valueCode = "";
+        String valueCodeDisplay = "";
+
+        if (practitionerRole.hasCode()) {
+            if (practitionerRole.getCodeFirstRep().hasCoding()) {
```

Comment:
Extract `codingFirstRep` into variable and it will simplify lines 42-47

Reply:
resolved